_posts/2020-11-09-introduction-to-github.md

```diff
@@ -0,0 +1,59 @@
+
+When working on a project, especially if you are working in a team, sharing the work you have done with each other is important. GitHub allows software developers to share their code and each person who is working on the project can do their own work without messing up the other persons progress.
+
+Say there are two people working on a project: Person 1 is the main coder who develops the base line code for everyone to work off of. Person 2 is a contributor that can also work on the code and provide feedback and maybe make some slight changes to what Person 1 has done. The basic workflow that would go into this two-person project would be as follows:
+  * **Person 1:** Creates working code for what the project calls for and pushes this code to GitHub
+  * **Person 2:** Fork the code onto their GitHub account and copy the code onto their machine. Make changes to the code. Push code to their GitHub. Open a pull request from the original code.
+  * **Person 1:** Look at Person 2’s pull request. If the code is good, approve the pull request and merge the changes into the main project
+
+This is just the very basics of what two people can do when working collaboratively through GitHub *(there is a lot of jargon in there that I will cover in the Version Control section of this post)*. But the key takeaway is that GitHub allows people to work collaboratively in a way that allows people to do their own work and not mess with each other’s progress.
+
+## The Importance of Version Control
+
+![version control]({{ site.url }}/assets/version_control.png)
+
+It is very important that you keep track of what you’ve done and what works when writing code. Version control goes hand-in-hand with collaborative coding. As previously mentioned, GitHub is great for collaborative projects because of its version control ability. If Person 2 were to work directly on the code that Person 1 was developing, that would break the code and ruin the whole project.
+
+GitHub allows for version control in several ways, some of which are detailed below. For continuity’s sake, I will be referencing the two-person project I detailed in the previous section:
+  * **The ability to fork a repository:** Person 2 can go to person 1’s GitHub profile and find the repository that has their project in it. They can then fork this repository to their GitHub account.
+    * This basically means that they copy the project to their GitHub account so they can use it as their own and not mess with what Person 1 is doing.
+
+
+  * **Pushing Code to GitHub:** if you are working on the command line this can be done through the command ‘git push’ followed by the name of the file that you would like to push to GitHub. This allows Person 2 to take whatever they were working on and put it onto GitHub for Person 1 to see.
+    * This step is not inherently about version control, but it is an important step that needs to be taken to maintain project continuity.
+
```

Comment:
Make note about 'git pull' as well. You have to pull stuff from GitHub once a new version gets added, or if you have to start working on a new repository.